src/inference/src/dev/core_impl.hpp

```diff
@@ -212,6 +212,11 @@ class CoreImpl : public InferenceEngine::ICore, public std::enable_shared_from_t
                                       const ov::Plugin& plugin,
                                       const ov::AnyMap& config) const;
 
+    std::string calculate_model_hash(const std::shared_ptr<ov::Model>& model,
+                                     const std::string& deviceFamily,
+                                     const ov::Plugin& plugin,
+                                     const ov::AnyMap& config) const;
```

Comment:
looks like this method does not have to be a class member. We can hide it as an internal implementation.